src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
```

Comment:
You have this check 4 times:
        if (index < 0 || index > size()) {
            throw new ArrayListIndexOutOfBoundsException("Index passed to the method is invalid");
        } else {
It's better to create separate method for this check. After it your methods will look like this:
    
    public T get(int index) {
        checkIndex(index);
        return elements[index];
    }

    public void set(T value, int index) {
        checkIndex(index);
        elements[index] = value;
    }

    public T remove(int index) {
        checkIndex(index);
        Object removeElement = elements[index];
        ...
    }

Reply:
Try to get how to create this method

 private ArrayListIndexOutOfBoundsException checkIndex(int index) {
        if (index < 0 || index > size) {
            throw new ArrayListIndexOutOfBoundsException("Index passed to the method is invalid");
        }
       return null;
    }